WPFHospitalEditor/Building2.xaml.cs

```diff
@@ -0,0 +1,139 @@
+using System.Text;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+using WPFHospitalEditor.MapObjectModel;
+
+namespace WPFHospitalEditor
+{
+    /// <summary>
+    /// Interaction logic for Building2.xaml
+    /// </summary>
+    public partial class Building2 : Window
+    {
+        AllMapObjects allMapObjects = new AllMapObjects();
+
+        public Building2()
+        {
+            InitializeComponent();
+            clearAll();
+            addObjectToCanvas(AllMapObjects.allSecondBuildingFirstFloorObjects);
+            displayLegend(AllMapObjects.allSecondBuildingFirstFloorObjects);
+        }
+
+        private void back_Click(object sender, RoutedEventArgs e)
+        {
+            clearAll();
+            HospitalMap window1 = new HospitalMap();
+            this.Close();
+            window1.ShowDialog();
+        }
+
+        private void addObjectToCanvas(List<MapObject> objectsToShow)
+        {
+            for (int i = 0; i < objectsToShow.Count; i++)
+            {
+                canvas.Children.Add(objectsToShow[i].rectangle);
+                canvas.Children.Add(objectsToShow[i].name);
+                canvas.Children.Add(objectsToShow[i].MapObjectDoor.rectangle);
+            }
+        }
+
+        private void floor_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            clearAll();
+            if (floor.SelectedIndex == 0)
+            {
+                addObjectToCanvas(AllMapObjects.allSecondBuildingFirstFloorObjects);
+                displayLegend(AllMapObjects.allSecondBuildingFirstFloorObjects);
+            }
+            else if (floor.SelectedIndex == 1)
+            {
+                addObjectToCanvas(AllMapObjects.allSecondBuildingSecondFloorObjects);
+                displayLegend(AllMapObjects.allSecondBuildingSecondFloorObjects);
+            }
+        }
+
+        private void clearAll()
+        {
+            canvas.Children.Clear();
+            if(legend1 != null && legend2 != null && legend3 != null)
+            {
+                legend1.Children.Clear();
+                legend2.Children.Clear();
+                legend3.Children.Clear();
+            }
+        }
+
+        private void displayLegend(List<MapObject> displayedMapObjects)
+        {
+            int itemsInOneRow = 0;
+            List<MapObjectType> mapObjectTypes = new List<MapObjectType>();
+            for(int i = 0; i < displayedMapObjects.Count; i++)
+            {
+                if (!mapObjectTypes.Contains(displayedMapObjects[i].MapObjectType))
+                {
+                    mapObjectTypes.Add(displayedMapObjects[i].MapObjectType);
+                    itemsInOneRow = organiseLegend(displayedMapObjects[i].MapObjectType, itemsInOneRow);
+                }
```

Comment:
To create a list of unique values, it is better to use a structure that takes care of that (map or set). Later, this structure can be easily converted to a list if needed.